Fix Qt deprecation warnings

The CI is showing warnings for deprecated methods related to [QAtomicInt](https://doc.qt.io/qt-5/qatomicinteger-obsolete.html).


```
2020-04-22T21:52:39.0313930Z [17/33] Building CXX object CMakeFiles/CCCoreLib.dir/src/NormalizedProgress.cpp.o
2020-04-22T21:52:39.0318640Z ../src/NormalizedProgress.cpp:86:97: warning: 'load' is deprecated: Use loadRelaxed [-Wdeprecated-declarations]
2020-04-22T21:52:39.0340880Z                         m_percent = static_cast<float>(totalPercentage) / totalSteps * static_cast<float>(m_counter->load());
2020-04-22T21:52:39.0346590Z                                                                                                                      ^
2020-04-22T21:52:39.0408610Z /usr/local/opt/qt5/lib/QtCore.framework/Headers/qbasicatomic.h:103:5: note: 'load' has been explicitly marked deprecated here
2020-04-22T21:52:39.0426300Z     QT_DEPRECATED_VERSION_X_5_14("Use loadRelaxed") T load() const noexcept { return loadRelaxed(); }
2020-04-22T21:52:39.0510330Z     ^
2020-04-22T21:52:39.0612280Z /usr/local/opt/qt5/lib/QtCore.framework/Headers/qglobal.h:364:45: note: expanded from macro 'QT_DEPRECATED_VERSION_X_5_14'
2020-04-22T21:52:39.0713050Z # define QT_DEPRECATED_VERSION_X_5_14(text) QT_DEPRECATED_X(text)
2020-04-22T21:52:39.0814470Z                                             ^
2020-04-22T21:52:39.0916910Z /usr/local/opt/qt5/lib/QtCore.framework/Headers/qglobal.h:294:33: note: expanded from macro 'QT_DEPRECATED_X'
2020-04-22T21:52:39.1018140Z #  define QT_DEPRECATED_X(text) Q_DECL_DEPRECATED_X(text)
2020-04-22T21:52:39.1128030Z                                 ^
2020-04-22T21:52:39.1229790Z /usr/local/opt/qt5/lib/QtCore.framework/Headers/qcompilerdetection.h:676:55: note: expanded from macro 'Q_DECL_DEPRECATED_X'
2020-04-22T21:52:39.1296310Z #    define Q_DECL_DEPRECATED_X(text) __attribute__ ((__deprecated__(text)))
2020-04-22T21:52:39.1329720Z                                                       ^
2020-04-22T21:52:39.1412150Z ../src/NormalizedProgress.cpp:90:15: warning: 'store' is deprecated: Use storeRelaxed [-Wdeprecated-declarations]
2020-04-22T21:52:39.1499710Z                         m_counter->store(0);
2020-04-22T21:52:39.1538160Z                                    ^
2020-04-22T21:52:39.1640080Z /usr/local/opt/qt5/lib/QtCore.framework/Headers/qbasicatomic.h:104:5: note: 'store' has been explicitly marked deprecated here
2020-04-22T21:52:39.1739380Z     QT_DEPRECATED_VERSION_X_5_14("Use storeRelaxed") void store(T newValue) noexcept { storeRelaxed(newValue); }
2020-04-22T21:52:39.1840700Z     ^
2020-04-22T21:52:39.1920950Z /usr/local/opt/qt5/lib/QtCore.framework/Headers/qglobal.h:364:45: note: expanded from macro 'QT_DEPRECATED_VERSION_X_5_14'
2020-04-22T21:52:39.2034320Z # define QT_DEPRECATED_VERSION_X_5_14(text) QT_DEPRECATED_X(text)
2020-04-22T21:52:39.2135570Z                                             ^
2020-04-22T21:52:39.2209240Z /usr/local/opt/qt5/lib/QtCore.framework/Headers/qglobal.h:294:33: note: expanded from macro 'QT_DEPRECATED_X'
2020-04-22T21:52:39.2269080Z #  define QT_DEPRECATED_X(text) Q_DECL_DEPRECATED_X(text)
2020-04-22T21:52:39.2278340Z                                 ^
2020-04-22T21:52:39.2341560Z /usr/local/opt/qt5/lib/QtCore.framework/Headers/qcompilerdetection.h:676:55: note: expanded from macro 'Q_DECL_DEPRECATED_X'
2020-04-22T21:52:39.2397420Z #    define Q_DECL_DEPRECATED_X(text) __attribute__ ((__deprecated__(text)))
2020-04-22T21:52:39.2400960Z                                                       ^
2020-04-22T21:52:39.2407490Z ../src/NormalizedProgress.cpp:98:13: warning: 'store' is deprecated: Use storeRelaxed [-Wdeprecated-declarations]
2020-04-22T21:52:39.2501940Z         m_counter->store(0);
2020-04-22T21:52:39.2536330Z                    ^
2020-04-22T21:52:39.2637710Z /usr/local/opt/qt5/lib/QtCore.framework/Headers/qbasicatomic.h:104:5: note: 'store' has been explicitly marked deprecated here
2020-04-22T21:52:39.2638400Z     QT_DEPRECATED_VERSION_X_5_14("Use storeRelaxed") void store(T newValue) noexcept { storeRelaxed(newValue); }
2020-04-22T21:52:39.2638800Z     ^
2020-04-22T21:52:39.2639700Z /usr/local/opt/qt5/lib/QtCore.framework/Headers/qglobal.h:364:45: note: expanded from macro 'QT_DEPRECATED_VERSION_X_5_14'
2020-04-22T21:52:39.2640230Z # define QT_DEPRECATED_VERSION_X_5_14(text) QT_DEPRECATED_X(text)
2020-04-22T21:52:39.2640660Z                                             ^
2020-04-22T21:52:39.2641530Z /usr/local/opt/qt5/lib/QtCore.framework/Headers/qglobal.h:294:33: note: expanded from macro 'QT_DEPRECATED_X'
2020-04-22T21:52:39.2641970Z #  define QT_DEPRECATED_X(text) Q_DECL_DEPRECATED_X(text)
2020-04-22T21:52:39.2642400Z                                 ^
2020-04-22T21:52:39.2643260Z /usr/local/opt/qt5/lib/QtCore.framework/Headers/qcompilerdetection.h:676:55: note: expanded from macro 'Q_DECL_DEPRECATED_X'
2020-04-22T21:52:39.2643760Z #    define Q_DECL_DEPRECATED_X(text) __attribute__ ((__deprecated__(text)))
2020-04-22T21:52:39.2644190Z                                                       ^
2020-04-22T21:52:39.2644570Z 3 warnings generated.
```

--- a/include/GenericProgressCallback.h
+++ b/include/GenericProgressCallback.h
@@ -7,10 +7,10 @@
 #include "CCConst.h"
 #include "CCCoreLib.h"
 
-class AtomicCounter;
-
 namespace CCCoreLib
 {
+	class AtomicCounter;
+
 	//! A generic progress indicator interface to notify algorithms progress to the client application
 	class CC_CORE_LIB_API GenericProgressCallback
 	{
@@ -104,9 +104,7 @@ namespace CCCoreLib
 		//! Percentage added to total progress value at each step
 		float m_percentAdd;
 
-		//! Current number of calls to 'oneStep'
-		/** Thread safe if CC_CORE_LIB is compiled with Qt.
-		**/
+		//! Current number of calls to 'oneStep' (thread safe)
 		AtomicCounter* m_counter;
 
 		//! associated GenericProgressCallback

--- a/src/NormalizedProgress.cpp
+++ b/src/NormalizedProgress.cpp
@@ -4,37 +4,13 @@
 #include "GenericProgressCallback.h"
 
 //system
+#include <atomic>
 #include <cassert>
 
-#ifdef CC_CORE_LIB_USES_QT_CONCURRENT
-
-//we use Qt for the atomic counter
-#include <QAtomicInt>
-
-class AtomicCounter : public QAtomicInt
-{
-};
-
-#else
-
-//we use a fake QAtomicInt
-
-//! Fake QAtomicInt
-/** \warning Not thread safe!
-**/
-class AtomicCounter
-{
-public:
-	AtomicCounter() : m_value(0) {}
-	inline int load() { return m_value; }
-	inline void store(int value) { m_value = value; }
-	inline int fetchAndAddRelaxed(int add) { int original = m_value; m_value += add; return original; }
-	int m_value;
-};
-
-#endif
-
 using namespace CCCoreLib;
+
+// Use a class "wrapper" to avoid having to include <atomic> in header
+class CCCoreLib::AtomicCounter : public std::atomic_uint {};
 
 NormalizedProgress::NormalizedProgress(	GenericProgressCallback* callback,
 										unsigned totalSteps,
@@ -42,7 +18,7 @@ NormalizedProgress::NormalizedProgress(	GenericProgressCallback* callback,
 	: m_percent(0)
 	, m_step(1)
 	, m_percentAdd(1.0f)
-	, m_counter(new AtomicCounter)
+	, m_counter( new AtomicCounter{} )
 	, progressCallback(callback)
 {
 	scale(totalSteps, totalPercentage);
@@ -50,10 +26,7 @@ NormalizedProgress::NormalizedProgress(	GenericProgressCallback* callback,
 
 NormalizedProgress::~NormalizedProgress()
 {
-	if (m_counter)
-	{
-		delete m_counter;
-	}
+	delete m_counter;
 }
 
 void NormalizedProgress::scale(	unsigned totalSteps,
@@ -109,7 +82,7 @@ bool NormalizedProgress::oneStep()
 		return true;
 	}
 
-	unsigned currentCount = static_cast<unsigned>(m_counter->fetchAndAddRelaxed(1)) + 1;
+	unsigned currentCount = m_counter->fetch_add( 1 ) + 1;
 	if ((currentCount % m_step) == 0)
 	{
 		m_percent += m_percentAdd;
@@ -126,7 +99,7 @@ bool NormalizedProgress::steps(unsigned n)
 		return true;
 	}
 
-	unsigned currentCount = static_cast<unsigned>(m_counter->fetchAndAddRelaxed(n)) + n;
+	unsigned currentCount = m_counter->fetch_add( n ) + n;
 	unsigned d1 = currentCount / m_step;
 	unsigned d2 = (currentCount + n) / m_step;
 

